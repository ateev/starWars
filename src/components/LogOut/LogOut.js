import React, { Component } from 'react';
import { deleteCookie } from '../../helpers/cookieHandler.js';
import { logoutUser } from '../../actions/userActions.js';
import { connect } from 'react-redux';
import './LogOut.css';

export class LogOut extends Component {
  logout = () => {
    this.context.store.dispatch(logoutUser());
  }
  render() {
    return (
        <div className="Log-out">
            <a onClick={this.logout} >Logout</a>
        </div>
    );
  }
}

LogOut.contextTypes = {
  store: React.PropTypes.object,
};

function mapStateToProps(state) {
  return {
    user: state.user,
  };
}

export default connect(mapStateToProps)(LogOut);
