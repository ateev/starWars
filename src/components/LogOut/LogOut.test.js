import React from 'react';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import { shallow, mount } from 'enzyme';
import LogOut from './LogOut.js';
import { Provider } from 'react-redux';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);


describe('<LogOut />', () => {
    it('renders without crashing', () => {
        const component = shallow(
            <Provider store={mockStore}>
                <LogOut />
            </Provider>
        );
        expect(component).toMatchSnapshot();
    });
})
