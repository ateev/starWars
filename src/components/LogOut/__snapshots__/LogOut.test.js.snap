exports[`<LogOut /> renders without crashing 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Connect(LogOut) />,
  "nodes": Array [
    <Connect(LogOut) />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Provider
        store={[Function]}>
        <Connect(LogOut) />
</Provider>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Provider {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": <Connect(LogOut) />,
          "store": [Function],
        },
        "refs": Object {},
        "state": null,
        "store": [Function],
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Connect(LogOut) />,
        "_debugID": 2,
        "_renderedOutput": <Connect(LogOut) />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Provider
    store={[Function]}>
    <Connect(LogOut) />
</Provider>,
}
`;
